CPAC/pipeline/schema.py

```diff
@@ -169,127 +212,278 @@
                     'summary': {
                         'method': str,
-                        'components': int
+                        'components': int,
+                        'filter': str,
                     },
+                    'threshold': str,
                     'tissues': [str],
                     'extraction_resolution': int
                 },
                 'CerebrospinalFluid': {
-                    'summary': str,
+                    'summary': Any(
+                        str, {'components': int, 'method': str}
+                    ),
                     'extraction_resolution': int,
                     'erode_mask': bool
                 },
                 'GlobalSignal': {'summary': str},
                 'PolyOrt': {'degree': int},
+                'WhiteMatter': {
+                    'erode_mask': bool,
+                    'extraction_resolution': Any(int, float, str),
+                    'summary': Any(
+                        str, {'components': int, 'method': str}
+                    ),
+                },
                 'Bandpass': {
                     'bottom_frequency': float,
                     'top_frequency': float
                 }  # how to check if [0] is > than [1]?
-            }],
-            'lateral_ventricles_mask': str,
-            Required('bandpass_filtering_order'): In({'After', 'Before'})
+            }]),
+            'lateral_ventricles_mask': Maybe(str),
+            'bandpass_filtering_order': Maybe(
+                In({'After', 'Before'}))
         },
     },
-    Required('amplitude_low_frequency_fluctuation'): {
-        Required('run'): bool,
+    'amplitude_low_frequency_fluctuation': {
+        'run': bool,
         'highpass_cutoff': [float],
         'lowpass_cutoff': [float],
     },
-    Required('voxel_mirrored_homotopic_connectivity'): {
-        Required('run'): bool,
+    'voxel_mirrored_homotopic_connectivity': {
+        'run': bool,
+        'symmetric_registration': {
+            'template_symmetric_brain_only': str,
+            'template_symmetric_brain_for_resample': str,
+            'template_symmetric_skull': str,
+            'template_symmetric_skull_for_resample': str,
+            'dilated_symmetric_brain_mask': str,
+            'dilated_symmetric_brain_mask_for_resample': str,
+            'FNIRT_pipelines': {
+                'config_file': str
+            },
+        },
     },
-    Required('regional_homogeneity'): {
-        Required('run'): bool,
-        'clusterSize': In({7, 19, 27}),
+    'regional_homogeneity': {
+        'run': bool,
+        'cluster_size': In({7, 19, 27}),
+    },
+    'post_processing': {
+        'spatial_smoothing': {
+            'run': Any(bool, [bool]),
+            'smoothing_method': In({'FSL', 'AFNI'}),
+            'fwhm': [int],
+            'smoothing_order': In({'Before', 'After'})
+        },
+        'z-scoring': {
+            'run': Any(bool, [bool]),
```

Comment:
I marked both of these `'run'` keys as either `bool` or a list of `bool`s because at least one regtest (regtest-2) has them both on/off: https://github.com/FCP-INDI/C-PAC/blob/8c287c30b83f85256c812343eb59fc708f3c6f93/CPAC/resources/configs/pipeline_config_regtest-2.yml#L206-L214

I think the refactored code expects a single boolean for each of them though.